tests/end-to-end/upgrade/pkg/tests/eventmesh-migration/helpers/application.go

```diff
@@ -0,0 +1,145 @@
+	return err
+}
+
+func WithAccessLabel(label string) ApplicationOption {
+	return func(application *appconnectorv1alpha1.Application) {
+		application.Spec.AccessLabel = label
+	}
+}
+
+func WithEventService(id string) ApplicationOption {
+	return func(application *appconnectorv1alpha1.Application) {
+		application.Spec.Services = append(application.Spec.Services,
+			appconnectorv1alpha1.Service{
+				ID:          id,
+				Name:        id,
+				DisplayName: "Some Events",
+				Description: "Application Service Class with Events",
+				Labels: map[string]string{
+					"connected-app": "app-name",
+				},
+				ProviderDisplayName: "provider",
+				Entries: []appconnectorv1alpha1.Entry{
+					{
+						Type: "Events",
+					},
+				},
+			})
+	}
+}
+
+func WaitForApplication(appConnector appconnector.Interface, messaging messagingclientv1alpha1.MessagingV1alpha1Interface, serving serving.Interface, name string, retryOptions ...retry.Option) error {
+	application, err := appConnector.ApplicationconnectorV1alpha1().Applications().Get(name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot get application: %+v", err)
+	}
+	if !application.Spec.SkipInstallation {
+		if err := WaitForChannel(messaging, name, integrationNamespace, retryOptions...); err != nil {
+			return fmt.Errorf("waiting for application failed: %+v, application: %+v", err, application)
+		}
+		if err := WaitForHttpSource(serving, name, integrationNamespace, retryOptions...); err != nil {
+			return fmt.Errorf("waiting for application failed: %+v, application: %+v", err, application)
+		}
+	}
+	return nil
+}
+
+func WaitForServiceInstance(serviceCatalog servicecatalog.Interface, name, namespace string, retryOptions ...retry.Option) error {
+	return retry.Do(
+		func() error {
+			sc, err := serviceCatalog.ServicecatalogV1beta1().ServiceInstances(namespace).Get(name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if sc.Status.ProvisionStatus != scv1beta1.ServiceInstanceProvisionStatusProvisioned {
+				return fmt.Errorf("service instance %v not provisioned yet: %v", name, sc.Status.ProvisionStatus)
+			}
+			return nil
+		}, retryOptions...)
+}
+
+func WaitForChannel(messaging messagingclientv1alpha1.MessagingV1alpha1Interface, name, namespace string, retryOptions ...retry.Option) error {
+	return retry.Do(
+		func() error {
+			ch, err := messaging.Channels(namespace).Get(name, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("cannot get channel: %+v", err)
+			}
+			if !ch.Status.IsReady() {
+				return fmt.Errorf("channel %v not ready", ch.Name)
+			}
+			return nil
+		}, retryOptions...)
+}
+
+func WaitForHttpSource(serving serving.Interface, name, namespace string, retryOptions ...retry.Option) error {
+	return retry.Do(
+		func() error {
+			ksvc, err := serving.ServingV1alpha1().Services(namespace).Get(name, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("cannot get HttpSource: %+v", err)
```

Comment:
errors are printed as string, so `%s` or `%v` is enough.

Reply:
removed +